store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -817,28 +817,79 @@ private Object getTargetMailbox(OperationContext octxt, Account authuser, boolea
         }
     }
 
-    public void logMessage(MimeMessage mm, String smtpHost, ItemId origMsgId, Collection<Upload> uploads, String replyType) {
+    public void logMessage(MimeMessage mm, Address[] rcptAddresses, String smtpHost, ItemId origMsgId, Collection<Upload> uploads, String replyType) {
         // Log sent message info
-        if (ZimbraLog.smtp.isInfoEnabled()) {
-            StringBuilder msg = new StringBuilder("Sending message");
-            if (smtpHost != null) {
-                msg.append(" to MTA at ").append(smtpHost);
-            }
-            try {
-                msg.append(": Message-ID=" + mm.getMessageID());
-            } catch (MessagingException e) {
-                msg.append(e);
-            }
-            if (origMsgId != null) {
-                msg.append(", origMsgId=" + origMsgId);
-            }
-            if (replyType != null) {
-                msg.append(", replyType=" + replyType);
+        if (!ZimbraLog.smtp.isInfoEnabled()) {
+            return;
+        }
+        StringBuilder msg = new StringBuilder("Sending message");
+        appendMsgMTA(msg, smtpHost);
+        appendMsgMessageID(msg, mm, origMsgId);
+        appendReplyType(msg, replyType);
+        appendUploads(msg, uploads);
+        appendSize(msg, mm);
+        appendEnvelopeFrom(msg, mEnvelopeFrom);
+        appendEnvelopeTo(msg, rcptAddresses);
```

Comment:
There are security and personal information disclosure. 
This may add Bcc address in logs. 
@plobbes  do you think it is good idea to log such information?

Reply:
Thanks @Prashantsurana for the comment.  In the first place, this fix is used by the system administrator (especially the ISP operator) to track down the message flow from the WebClient action through the MBS to the external MTA. On the MTA side, for example Postfix, all the RCPT TO addresses including Bcc have been already logged.

Reply:
As @ykomiyam mentioned, bcc is logged as part of RCPT TO for MTAs. Logging these as part of the 'to' addresses on the mailbox servers does not introduce a new/additional exposure beyond what we already do in the product.